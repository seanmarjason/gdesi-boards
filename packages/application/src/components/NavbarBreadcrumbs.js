import * as React from 'react';
import { styled } from '@mui/material/styles';
import Typography from '@mui/material/Typography';
import Breadcrumbs, { breadcrumbsClasses } from '@mui/material/Breadcrumbs';
import NavigateNextRoundedIcon from '@mui/icons-material/NavigateNextRounded';
import Link from '@mui/material/Link';

export default function NavbarBreadcrumbs({ nav=['']}) {

  const lastNav = nav.slice(nav.length-1)
  const prevNav = nav.slice(0, -1)

  return (
    <Breadcrumbs
      aria-label="breadcrumb"
      separator={<NavigateNextRoundedIcon fontSize="small" />}
    >
      { prevNav.map((link, index) =>
          <Typography variant="body1" key={`breadcrumb-${index}`}>
              <Link href={`${'../'.repeat(nav.length - index)}${link.toLowerCase().replace(/ /g,"-")}`}>
                {link}
              </Link>
          </Typography>
      )}
      <Typography variant="body1" sx={{ color: 'text.primary', fontWeight: 600 }}>
        {lastNav}
      </Typography>
    </Breadcrumbs>
  );
}
