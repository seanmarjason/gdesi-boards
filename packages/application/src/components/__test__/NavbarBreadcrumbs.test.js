import { render } from '@testing-library/react'
import NavbarBreadcrumbs from '../NavbarBreadcrumbs';

test('NavbarBreadcrumbs renders correctly', () => {
  const { container } = render(<NavbarBreadcrumbs />)
  expect(container).toMatchSnapshot()
})
