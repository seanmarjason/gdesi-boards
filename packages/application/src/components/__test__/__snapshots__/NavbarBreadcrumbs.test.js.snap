// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarBreadcrumbs renders correctly 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1d2i4tu-MuiTypography-root"
        />
      </li>
    </ol>
  </nav>
</div>
`;
